encoding/protobq/marshal.go

```diff
@@ -66,13 +66,23 @@ func (o MarshalOptions) marshalMessage(msg protoreflect.Message) (map[string]big
 				returnErr = errMarshal
 				return false
 			}
+			if m, ok := column.(map[string]bigquery.Value); ok && len(m) == 0 {
+				// don't set anything for empty records
+				return true
+			}
 			result[string(field.Name())] = column
 		}
 		return true
 	})
 	if returnErr != nil {
 		return nil, returnErr
 	}
+	if o.Schema.UseOneofFields {
+		for i := 0; i < msg.Descriptor().Oneofs().Len(); i++ {
+			oneof := msg.Descriptor().Oneofs().Get(i)
+			result[string(oneof.Name())] = string(msg.WhichOneof(oneof).Name())
```

Comment:
What if the oneof is not set here? What will `Name()` return?

Reply:
Oooh! Nice catch. That's a nil pointer. 😁

Reply:
:smile_cat: I felt a disturbance in the force